### PR TITLE
autotest: added --no-rcin option to sim_vehicle.py

### PR DESCRIPTION
**Pull request by tridge**
disables mavproxy RC in emulation

### CODE REVIEW

File: Tools/autotest/sim_vehicle.py [hunk 1]
```diff
@@ -985,6 +985,9 @@ def generate_frame_help():
                  default=0,
                  type=int,
                  help="mavproxy module debug")
+group.add_option("", "--no-rcin",
```
Comment (peterbarker) [inline Tools/autotest/sim_vehicle.py]: This really ought to be `--no-mavproxy-rc`.  Nothing stops something else connecting to the SITL RC-in port, right?

Comment (tridge) [reply Tools/autotest/sim_vehicle.py]: true, but not really likely to be anything else

**Comment by peterbarker** [comment]
On Wed, 4 Dec 2019, Andrew Tridgell wrote:

> In Tools/autotest/sim_vehicle.py:
> 
> > @@ -985,6 +985,9 @@ def generate_frame_help():
>                   default=0,
>                   type=int,
>                   help="mavproxy module debug")
> +group.add_option("", "--no-rcin",
> 
> true, but not really likely to be anything else

You'd think - but then people did add that "--no-mavproxy" option, too :-)

**Comment by Hwurzburg** [comment]
just tested in sitl on plane.... default params (-w)....so throttle failsafe was enabled...all arming checks...
it armed and flew when switched to mode 13 (taekoff)....I would have thought I would have had to change something since a real plane would be in failsafe with no rc unless throttle fs was disabled, correct?

**Comment by WickedShell** [comment]
The way SITL works is that just because nothing is sending RC commands doesn't mean the simmed backend failed. IE I use the SITL binary without MAVProxy in the loop at all, and I have healthy (if constant) RC values according to the aircraft. If I want to see bad RC according to the sim I have to actually set `SIM_RC_FAIL`. So this could definitely be working as is. To test you'd need to actually try and send an RC (but not MAVLink override command) from mavproxy and watch the reported RC/PWM values.

**Comment by Hwurzburg** [comment]
Thanks...tridge did this so that we could test rc-less setups to make sure that any config req'd was correct for gcs only control with no rc attached...but it seems that the most accurate way would be to just set the SIM_RC_FAIL to 1...correct?

**Comment by WickedShell** [comment]
Yeah, then turn off `THR_FAILSAFE`. Depending on your GCS you will still get a warning here, as the vehicle thinks its RC is unhealthy, even though we don't require it. As long as the GCS isn't sending the RC values it should be pretty much fine, I do all my SITL work without sending anything in via the simmed RC inputs (usually without setting the fail bit just to keep the warning away) so you shouldn't have to many problems.

**Pull request merged**